packages/svelte/src/compiler/phases/2-analyze/css/css-analyze.js

```diff
@@ -118,21 +141,35 @@ const validation_visitors = {
 			}
 
 			const complex_selector = node.prelude.children[0];
-			const relative_selector = complex_selector.children[complex_selector.children.length - 1];
+			const global_selector = complex_selector.children.find((r, selector_idx) => {
+				const idx = r.selectors.findIndex(is_global_block_selector);
+				if (idx === 0) {
+					if (r.selectors.length > 1 && selector_idx === 0 && node.metadata.parent_rule === null) {
+						e.css_global_block_invalid_modifier_start(r.selectors[1]);
+					}
+					return true;
+				} else if (idx !== -1) {
+					e.css_global_block_invalid_modifier(r.selectors[idx]);
+				}
+			});
 
-			if (relative_selector.selectors.length > 1) {
-				e.css_global_block_invalid_modifier(
-					relative_selector.selectors[relative_selector.selectors.length - 1]
-				);
+			if (!global_selector) {
+				throw new Error('Internal error: global block without :global selector');
 			}
 
-			if (relative_selector.combinator && relative_selector.combinator.name !== ' ') {
-				e.css_global_block_invalid_combinator(relative_selector, relative_selector.combinator.name);
+			if (global_selector.combinator && global_selector.combinator.name !== ' ') {
+				e.css_global_block_invalid_combinator(global_selector, global_selector.combinator.name);
 			}
 
 			const declaration = node.block.children.find((child) => child.type === 'Declaration');
 
-			if (declaration) {
+			if (
+				declaration &&
+				// :global { color: red; } is invalid, but foo :global { color: red; } is valid
```

Comment:
why this change? I can't think of a good reason to allow declarations inside a `:global {...}` block

Reply:
We need that to satisfy the Tailwind case
```css
main :global {
  @apply bg-red-800
}
/* Tailwind will turn this into something like  */
main :global {
  --tw-bg-opacity: 1;
  background-color: rgb(219 234 254 / var(--tw-bg-opacity));
}
```
Also it's just overly restrictive to disallow it when the resulting CSS is perfectly fine

Reply:
To me it's more like tailwind is poking holes into our made-up global semantics and us robustifying it as a result